stm32-modules/heater-shaker/firmware/system/system_hardware.c

```diff
@@ -60,3 +90,149 @@ asm volatile (
   : "r" (*sysmem_boot_loc)
   : "memory"  );
 }
+
+bool system_hardware_setup_led(void) {
+  uint32_t ulNotificationValue;
+  const TickType_t xMaxBlockTime = pdMS_TO_TICKS( 100 );
+  HAL_StatusTypeDef status;
+
+  configASSERT( xTaskToNotify == NULL );
+  xTaskToNotify = xTaskGetCurrentTaskHandle();
+  status = HAL_I2C_Mem_Write_IT(&I2cHandle, ((uint16_t)I2C_ADDRESS)<<1, (uint16_t)SHUTDOWN_REGISTER,
+    (uint16_t)REGISTER_SIZE, (uint8_t*)ShutdownBuffer, (uint16_t)(sizeof(ShutdownBuffer)));
+  ulNotificationValue = ulTaskNotifyTake( pdTRUE, xMaxBlockTime );
+
+  if ((status == HAL_OK) && (ulNotificationValue == 1) && (I2cHandle.State == HAL_I2C_STATE_READY)) {
```

Comment:
[nit] if we make this return early on error, there will be fewer nested `if`s and it'll be a bit easier to read